src/levelset/levelSetSegmentationObject.cpp

```diff
@@ -2575,7 +2575,6 @@ LevelSetCellLocation LevelSetSegmentationObject::fillCellGeometricNarrowBandLoca
     } else if (cellUnsigendValue <= m_narrowBandSize) {
         cellLocation = LevelSetCellLocation::NARROW_BAND_DISTANCE;
     }
-    assert(cellLocation != LevelSetCellLocation::UNKNOWN);
```

Comment:
This means that there will be cells that are inside the narrow band but tagged as LevelSetCellLocation::UNKNOWN? I don't think this will work.

Reply:
The cell location can be unknown because it's evaluated later. It happens in one of the new tests I've created. The removal of the assert is safe because whenever this function is used (fillCellGeometricNarrowBandLocationCache), the scenario of returning an unknown location is always considered into account.

Reply:
One of the places the function is used is here:
```
void LevelSetSegmentationObject::fillCartesianCellZoneCache()
{
        ...

        // Fill location cache for cells geometrically inside the narrow band
        LevelSetCellLocation cellLocation = fillCellGeometricNarrowBandLocationCache(cellId);

        // Track intersected cells
        if (cellLocation == LevelSetCellLocation::NARROW_BAND_INTERSECTED) {
            intersectedCellIds.push_back(cellId);
        }

        // Add unprocessed neighbours of cells inside the narrow band to the process list
        if (cellLocation != LevelSetCellLocation::UNKNOWN) {
            auto neighProcessor = [&processList, &alreadyProcessed](long neighId, int layer) {
                BITPIT_UNUSED(layer);

        ...
}

In this case cellLocation can be UNKNOWN and the meaning is "I don't know where the cell is, but is certainly outside the narrowband band". If the function returns UNKNOWN also for cells inside the narrow band (but that cannot be identified as such at that stage), which portion of the code will add the neighbours of those cells (cells inside the narrow band but whose location is UNKNOWN) to the process list?

Reply:
Answered bellow